django/db/models/expressions.py

```diff
@@ -56,10 +56,12 @@ class Combinable:
     def _combine(self, other, connector, reversed):
         if not hasattr(other, 'resolve_expression'):
             # everything must be resolvable to an expression
-            if isinstance(other, datetime.timedelta):
-                other = DurationValue(other, output_field=fields.DurationField())
-            else:
-                other = Value(other)
+            output_field = (
+                fields.DurationField()
+                if isinstance(other, datetime.timedelta) else
+                None
+            )
+            other = Value(other, output_field=output_field)
```

Comment:
We'll also be able to simplify this to the following the #11359 is merged

```suggestion
            other = Value(other)
```